cmd/kubeadm/app/master/manifests.go

```diff
@@ -0,0 +1,302 @@
+		},
+	}
+}
+
+func certsVolumeMount() api.VolumeMount {
+	return api.VolumeMount{
+		Name:      "certs",
+		MountPath: "/etc/ssl/certs",
+	}
+}
+
+func k8sVolume(s *kubeadmapi.KubeadmConfig) api.Volume {
+	return api.Volume{
+		Name: "pki",
+		VolumeSource: api.VolumeSource{
+			HostPath: &api.HostPathVolumeSource{Path: s.EnvParams["kubernetes_dir"]},
+		},
+	}
+}
+
+func k8sVolumeMount() api.VolumeMount {
+	return api.VolumeMount{
+		Name:      "pki",
+		MountPath: "/etc/kubernetes/",
+		ReadOnly:  true,
+	}
+}
+
+func componentResources(cpu string) api.ResourceRequirements {
+	return api.ResourceRequirements{
+		Requests: api.ResourceList{
+			api.ResourceName(api.ResourceCPU): resource.MustParse(cpu),
+		},
+	}
+}
+
+func componentProbe(port int, path string) *api.Probe {
+	return &api.Probe{
+		Handler: api.Handler{
+			HTTPGet: &api.HTTPGetAction{
+				Host: "127.0.0.1",
+				Path: path,
+				Port: intstr.FromInt(port),
+			},
+		},
+		InitialDelaySeconds: 15,
+		TimeoutSeconds:      15,
+	}
+}
+
+func componentPod(container api.Container, volumes ...api.Volume) api.Pod {
+	return api.Pod{
+		TypeMeta: unversioned.TypeMeta{
+			APIVersion: "v1",
+			Kind:       "Pod",
+		},
+		ObjectMeta: api.ObjectMeta{
+			Name:      container.Name,
+			Namespace: "kube-system",
+			Labels:    map[string]string{"component": container.Name, "tier": "control-plane"},
+		},
+		Spec: api.PodSpec{
+			Containers:  []api.Container{container},
+			HostNetwork: true,
+			Volumes:     volumes,
+		},
+	}
+}
+
+func getComponentCommand(component string, s *kubeadmapi.KubeadmConfig) (command []string) {
+	baseFlags := map[string][]string{
+		apiServer: {
+			"--address=127.0.0.1",
+			"--etcd-servers=http://127.0.0.1:2379",
+			"--admission-control=NamespaceLifecycle,LimitRanger,ServiceAccount,PersistentVolumeLabel,DefaultStorageClass,ResourceQuota",
+			"--service-cluster-ip-range=" + s.InitFlags.Services.CIDR.String(),
+			"--service-account-key-file=" + pkiDir + "/apiserver-key.pem",
+			"--client-ca-file=" + pkiDir + "/ca.pem",
+			"--tls-cert-file=" + pkiDir + "/apiserver.pem",
+			"--tls-private-key-file=" + pkiDir + "/apiserver-key.pem",
+			"--token-auth-file=" + pkiDir + "/tokens.csv",
+			"--secure-port=443",
+			"--allow-privileged",
+		},
+		controllerManager: {
+			// TODO(phase1+): consider adding --address=127.0.0.1 in order to not expose the cm port to the rest of the world
+			"--leader-elect",
+			"--master=127.0.0.1:8080",
+			"--cluster-name=" + DefaultClusterName,
+			"--root-ca-file=" + pkiDir + "/ca.pem",
+			"--service-account-private-key-file=" + pkiDir + "/apiserver-key.pem",
+			"--cluster-signing-cert-file=" + pkiDir + "/ca.pem",
+			"--cluster-signing-key-file=" + pkiDir + "/ca-key.pem",
+			"--insecure-experimental-approve-all-kubelet-csrs-for-group=system:kubelet-bootstrap",
+		},
+		scheduler: {
+			// TODO(phase1+): consider adding --address=127.0.0.1 in order to not expose the scheduler port to the rest of the world
```

Comment:
I really think this should be fixed. It would be strange to advertise we're setting up the cluster securely while exposing controller-manager and scheduler to the rest of the world if the apiserver is accessible to the outside